packages/mendel-pipeline/src/registry/client.js

```diff
@@ -110,7 +104,11 @@ class MendelOutletRegistry {
             return positives.some(g => g.match(id)) &&
                 negatives.every(g => g.match(id));
         })
-        .map(id => this.getEntry(id));
+        .map(id => this.getEntry(id))
+        .filter(({variation}) => {
+            return !excludeVariational ||
+                variation === this._options.baseConfig.dir;
+        });
```

Comment:
>  I was under the impression that picking up from all variation dirs was the proper behavior. It might be breaking something, but Not in full example, at least.

Not true. 
https://github.com/yahoo/mendel/blob/master/examples/full-example/.mendelrc#L135
https://github.com/yahoo/mendel/blob/e46e7bf1c67d28c12f7566432e58a4406bbb9864/packages/mendel-pipeline/src/bundles/default-generator.js#L10

Imposing that you create a file in base variation for "entries" is not a bad one. Btw. `walk` will take care of variations. 
https://github.com/yahoo/mendel/blob/e46e7bf1c67d28c12f7566432e58a4406bbb9864/packages/mendel-pipeline/src/registry/client.js#L124

In summary, there may be no harm in walking in variational dir too but there is a strong chance it will do redundant walking.